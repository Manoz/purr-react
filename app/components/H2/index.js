import styled from 'styled-components';

const H2 = styled.h2`
  margin-top: 1.2em;
  font-weight: 500;
  font-size: 2rem;
  line-height: 1.09375;
  letter-spacing: .011em;
`;

export default H2;
