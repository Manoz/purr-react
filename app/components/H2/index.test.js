import React from 'react';
import { shallow } from 'enzyme';

import H2 from './index';

describe('<H2 />', () => {
  it('should render a <h2> tag', () => {
    const wrapper = shallow(<H2 />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const wrapper = shallow(<H2 id={id} />);
    expect(wrapper.prop('id')).toEqual(id);
  });

  it('should not adopt an invalid attribute', () => {
    const wrapper = shallow(<H2 attribute="test" />);
    expect(wrapper.prop('attribute')).toEqual('test');
  });
});
