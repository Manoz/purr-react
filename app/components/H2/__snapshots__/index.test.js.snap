// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H2 /> should render a <h2> tag 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "H2-zasn0o-0",
        "isStatic": true,
        "rules": Array [
          "margin-top:1.2em;font-weight:500;@media (min-width:768px){font-size:24px;line-height:1.20849;letter-spacing:.015em;}@media (min-width:992px){font-size:28px;line-height:1.1073;letter-spacing:.012em;}@media (min-width:1200px){font-size:32px;line-height:1.09375;letter-spacing:.011em;}",
        ],
      },
      "displayName": "H2",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "H2-zasn0o-0",
      "target": "h2",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
